### PR TITLE
[14.0] sale_configurator: visual improvements relative to options in backend + report changes

### PR DESCRIPTION
**Pull request by kevinkhao**
add visual distinction to option lines for sale order lines + invoice lines
add visual distinction to option lines on invoice form view
rename CSS class to generic configurator_option_padding

### CODE REVIEW

File: sale_configurator_option/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+from odoo import api, models
+
+from odoo.addons.sale_configurator_base.models.sale import update_attrs
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move"
+
+    @api.model
```
Comment (sebastienbeau) [inline sale_configurator_option/models/account_move.py:14]: Why do we need a dynamic view here? Why not using pure xml?

**Review by sebastienbeau — CHANGES_REQUESTED** [review]
@kevinkhao can you move the report into the sale_configurator_base module ?
Indeed the we just need to apply this formating for all child line (whatever is an option or something else)

File: sale_configurator_option/models/account_move_line.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends("sale_line_ids")
```
Comment (sebastienbeau) [inline sale_configurator_option/models/account_move_line.py]: I think we should add the concept of parent_id/child_ids here. And fill this field when we create the invoice

File: sale_configurator_option/models/account_move.py (hunk 1 above)
Comment (kevinkhao) [reply sale_configurator_option/models/account_move.py:14]: I think this is related to the ir.ui.view part and things like _get_sale_line_tree_item, _get_sale_line_form_item, which I don't fully understand

File: sale_configurator_base/templates/sale_report_templates.xml [hunk 3]
```diff
@@ -105,8 +100,18 @@
         <xpath expr="//td[@name='td_subtotal']" position="replace">
             <td name="td_subtotal" class="text-right o_price_total">
                 <t
-                    t-if="not line.child_ids and not line.report_empty_parent and line.parent_id"
-                >
+                    t-set="is_child"
+                    t-value="line.parent_id and (not line.child_ids) and (not line.report_line_is_empty_parent)"
```
Comment (sebastienbeau) [inline sale_configurator_base/templates/sale_report_templates.xml]: a line with a parent can not have child so testing line.parent_id is enough (same for report_line_is_empty_parent this have no meaning for child line

**Pull request merged**